### ISSUE TITLE
Option to check what settings were used to create the filesystem

### ISSUE THREAD

**Issue by daci12345** [mhx/dwarfs#127]
An option to dwarfs maybe dwarfsck to check out what compression settings were used to create a dwarfs file.

**Comment by mhx** [comment]
Agreed, that would be useful. I've actually wanted this myself in the past. The thing that stopped me from implementing it right away is that you can make changes to the file system after it has initially been created (e.g. change compression settings, append/remove headers, maybe other things in the future).

I'm just thinking that what you'd really need is a history of commands (including `mkdwarfs` versions) that was used to create/modify the file system. Let me know what you think.

**Comment by mhx** [comment]
Sneak preview of how this is going to look with v0.8.0 once that's released:

```
% ./dwarfsck wav-rewritten.dwarfs 
DwarFS version 2.5 [2]
History:
  1: [2023-12-16 23:33:59] libdwarfs v0.7.2-293-g126eb72-dirty (mhx/categorizer) on Linux-6.1.46-gentoo-dist [x86_64], Clang 17.0.6
     args: ./mkdwarfs -i ../build-clang/wav -o wav.dwarfs --categorize=pcmaudio,incompressible -l9 -C pcmaudio/waveform::flac:level=2 -C incompressible::null -W pcmaudio/waveform::20 --order pcmaudio/waveform::path --force -L 4g
  2: [2023-12-16 23:36:39] libdwarfs v0.7.2-293-g126eb72-dirty (mhx/categorizer) on Linux-6.1.46-gentoo-dist [x86_64], Clang 17.0.6
     args: ./mkdwarfs --recompress -i wav.dwarfs -o wav-rewritten.dwarfs --force -C pcmaudio/waveform::flac:level=7 -C incompressible::null
[...]
```

There will be a full history of every mutating operation of a DwarFS image by default, but you can disable the feature with the `--no-history` flag if so desired.

**Comment by mhx** [comment]
If you want to play with the new features, the [CI build workflow](https://github.com/mhx/dwarfs/actions/workflows/build.yml) now automatically uploads the binary artifacts (universal binaries as well as the regular binary tarball) for each build. These are definitely not "release quality", but it's still good to give them wider exposure to find bugs early.

**Comment by mhx** [comment]
[Fixed in v0.8.0.](https://github.com/mhx/dwarfs/releases/tag/v0.8.0)
